tools/perspective-scripts/workspace.mjs

```diff
@@ -0,0 +1,53 @@
+// ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+// ┃ ██████ ██████ ██████       █      █      █      █      █ █▄  ▀███ █       ┃
+// ┃ ▄▄▄▄▄█ █▄▄▄▄▄ ▄▄▄▄▄█  ▀▀▀▀▀█▀▀▀▀▀ █ ▀▀▀▀▀█ ████████▌▐███ ███▄  ▀█ █ ▀▀▀▀▀ ┃
+// ┃ █▀▀▀▀▀ █▀▀▀▀▀ █▀██▀▀ ▄▄▄▄▄ █ ▄▄▄▄▄█ ▄▄▄▄▄█ ████████▌▐███ █████▄   █ ▄▄▄▄▄ ┃
+// ┃ █      ██████ █  ▀█▄       █ ██████      █      ███▌▐███ ███████▄ █       ┃
+// ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
+// ┃ Copyright (c) 2017, the Perspective Authors.                              ┃
+// ┃ ╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌ ┃
+// ┃ This file is part of the Perspective library, distributed under the terms ┃
+// ┃ of the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0). ┃
+// ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+import fs from "node:fs";
+import url from "node:url";
+import path from "node:path";
+
+const __dirname = url.fileURLToPath(new URL(".", import.meta.url)).slice(0, -1);
+const workspaceRoot = path.normalize(path.join(__dirname, "..", ".."));
+const rustTargetDir = path.join(workspaceRoot, "rust", "target");
+const rustWheelsDir = path.join(rustTargetDir, "wheels");
+
+const memoize = (f) => {
+    let val = undefined;
+    return () => {
+        if (typeof val !== "undefined") return val;
+        val = f();
+        return val;
+    };
+};
+
+export function getWorkspaceRoot() {
+    return workspaceRoot;
+}
+export function getRustTargetDir() {
+    return rustTargetDir;
+}
+export function getRustWheelsDir() {
+    return rustWheelsDir;
+}
+export function getEmscriptenWheelPath() {
+    const pspVersion = getWorkspacePackageJson().version;
```

Comment:
I'm not sure, is the workspace package.json an OK place to get the version of the emscripten wheel? Or should this read from rust/perspective-python/Cargo.toml? If they're meant to be kept in sync, it shouldn't matter

Reply:
huh, interesting. maybe should do like this, and read it from perspective-python's package.json https://github.com/finos/perspective/pull/2739/files#diff-fbc295126e0b0b6fdc48b0a7f73d15b69d7cbb25140f65f0453200c1b341f809L28-R32

Reply:
We update all `package.json` files at once, so the "local" one is the right place to fethc it generally, except the scripts in `perspective-scripts` don't live in a module, so the workspace `package.json` is the only option.